vendor/github.com/softlayer/softlayer-go/datatypes/container.go

```diff
@@ -2140,7 +2140,7 @@ type Container_Network_CdnMarketplace_Configuration_Input struct {
 	Protocol *string `json:"protocol,omitempty" xmlrpc:"protocol,omitempty"`
 
 	// no documentation yet
-	RespectHeaders *string `json:"respectHeaders,omitempty" xmlrpc:"respectHeaders,omitempty"`
+	RespectHeaders *bool `json:"respectHeaders,omitempty" xmlrpc:"respectHeaders,omitempty"`
```

Comment:
Instead of changing the type can we change the type of respectHeaders to string and pass only two values true or false from schema?

Reply:
https://cloud.ibm.com/docs/infrastructure/CDN?topic=CDN-cdn-api-reference#createdomainmapping as per the above link, the respects headers need boolean value to be set.

Reply:
yes as per above link we need to pass boolean but it not valid to change the type in a vendor project..After little investigating from my end we can achieve this even without changing the type in vendor file..

Move back the type to string in vendor code (RespectHeaders *string `json:"respectHeaders,omitempty" xmlrpc:"respectHeaders,omitempty"`)
And in resource when user pass true or false send a string of "1" or "0"
```
respectheaders := d.Get("respectheaders").(bool)
var rHeader = "0"
if respectheaders {
     rHeader = "1"
}
And in createDomainMapping and updateDomainMapping pass the respectHeader as below..I tested the code its working as expected...
------
HttpsPort:                sl.Int(httpsport),
OriginType:              sl.String(origintype),
Header:                    sl.String(header),
RespectHeaders:     sl.String(rHeader),
------- 
```